cmd/daemon/serve.go

```diff
@@ -122,10 +121,19 @@ func ServePublic(r driver.Registry, wg *sync.WaitGroup, cmd *cobra.Command, args
 	}
 
 	certs := c.GetTSLCertificatesForPublic()
-	server := graceful.WithDefaults(&http.Server{
-		Handler:   handler,
-		TLSConfig: &tls.Config{Certificates: certs, MinVersion: tls.VersionTLS12},
-	})
+	var server *http.Server
+	if tracer := r.Tracer(ctx); tracer.IsLoaded() {
```

Comment:
Similar here. Also, is there a reason why the handler can't be a function that we can pass to `n.Use()`? I think that would be less code and easier to handle. All you have to do is to implement a function that follows the interface of the argument to `n.Use`